e3nn/util/dataset/molecules.py

```diff
@@ -149,3 +149,135 @@ def download(self):
 
         self.files = sorted(glob.glob(os.path.join(xyz_path, "*.xyz")))
         assert len(self.files) == 133885
+
+
+
+import torch
+
+import os
+from functools import partial
+from itertools import zip_longest
+
+from pymatgen.core.structure import Molecule
+from pymatgen.core.periodic_table import Element
+from torch_geometric.data import Dataset
+
+
+def to_one_hot(ind_arr, n_classes):
+    one_hot = torch.zeros((len(ind_arr), n_classes))
+    one_hot[torch.arange(len(ind_arr)), ind_arr] = 1.
+    return one_hot
+
+
+# function that constructs graph given 
+def molecule_from_file_to_graph_sphere_connectivity_5_no_self_interactions(molecule_file, y):
+    return molecule_from_file_to_graph_base(molecule_file, y, connectivity_func=connections_within_sphere_5_no_self_interactions)
+
+def connections_within_sphere_5_no_self_interactions(mol):
+    return connections_within_sphere(mol, r_cut=5.0, self_interaction=False)
+
+def connections_within_sphere(mol, r_cut, self_interaction):
+    edge_index_transposed = torch.tensor([[origin_site_idx, neighbor_site.index] 
+                                         for origin_site_idx, site in enumerate(mol.sites) 
+                                         for neighbor_site in mol.get_neighbors(site, r_cut)
+                                         ], dtype=torch.int64) 
+    if self_interaction:
+        self_interaction_indices = torch.arange(len(mol), dtype=torch.int64).view(len(mol), 1).repeat(1,2)
+        edge_index_transposed = torch.cat((edge_index_transposed, self_interaction_indices), dim=0)
+    return edge_index_transposed.t()
+
+def molecule_from_file_to_graph_base(molecule_file, y, connectivity_func):
+    mol = Molecule.from_file(molecule_file)
+
+    x = torch.tensor(mol.atomic_numbers) # further processing for features x is assumed to happen in transform
+    edge_index = connectivity_func(mol)  # not necessarily sphere
+    edge_attr = None
+    y = y
+    pos = torch.from_numpy(mol.cart_coords)
+
+    return Data(x, edge_index, edge_attr, y, pos)
+
+
+# transform function supplied to pytorch-geometric Dataset should accept as an input and produce as an output Data object (graph) only  
+def molecule_graph_transform_one_hot_period_group_5_18(graph):
+    return molecule_graph_transform_base(graph, featurizer=featurizer_one_hot_period_group_5_18)
+
+def featurizer_one_hot_period_group_5_18(graph):
```

Comment:
Again, I think this function is too specific -- it would however make an excellent test case.